py/tools/py/src/pth.rs

```diff
@@ -48,13 +32,90 @@ impl PthFile {
 
         let mut line = String::new();
         while reader.read_line(&mut line).unwrap() > 0 {
-            let entry = Path::new(prefix).join(Path::new(line.trim()));
+            let entry: PathBuf;
+            if self.prefix.is_some() {
+                entry = Path::new(self.prefix.as_deref().unwrap()).join(Path::new(line.trim()));
+            } else {
+                entry = PathBuf::from(line.trim());
+            }
             line.clear();
-            writeln!(writer, "{}", entry.to_string_lossy())
-                .into_diagnostic()
-                .wrap_err("Unable to write new .pth file entry with prefix")?;
+            if entry.file_name().is_some_and(|x| x == "site-packages") {
+                let src_dir = dest.join(entry).canonicalize().unwrap();
+                create_symlinks(&src_dir, &src_dir, &dest)?;
+            } else {
+                writeln!(writer, "{}", entry.to_string_lossy())
+                    .into_diagnostic()
+                    .wrap_err("Unable to write new .pth file entry")?;
+            }
         }
 
         Ok(())
     }
 }
+
+fn create_symlinks(dir: &Path, root_dir: &Path, dst_dir: &Path) -> miette::Result<()> {
+    // Create this directory at the destination.
+    let tgt_dir = dst_dir.join(dir.strip_prefix(root_dir).unwrap());
+    std::fs::create_dir_all(&tgt_dir)
+        .into_diagnostic()
+        .wrap_err(format!(
+            "unable to create parent directory for symlink: {}",
+            tgt_dir.to_string_lossy()
+        ))?;
+
+    // Recurse.
+    let read_dir = fs::read_dir(dir).into_diagnostic().wrap_err(format!(
+        "unable to read directory {}",
+        dir.to_string_lossy()
+    ))?;
+    for entry in read_dir {
+        let entry = entry.into_diagnostic().wrap_err(format!(
+            "unable to read directory entry {}",
+            dir.to_string_lossy()
+        ))?;
+        let path = entry.path();
+        if path.is_dir() {
+            create_symlinks(&path, root_dir, dst_dir)?;
+        } else if dir != root_dir || entry.file_name() != "__init__.py" {
+            create_symlink(&entry, root_dir, dst_dir)?;
+        }
+    }
+    Ok(())
+}
+
+fn create_symlink(e: &DirEntry, root_dir: &Path, dst_dir: &Path) -> miette::Result<()> {
+    let tgt = e.path();
+    let link = dst_dir.join(tgt.strip_prefix(root_dir).unwrap());
+    if link.exists() && link.file_name().is_some_and(|x| x == "__init__.py") && is_same_file(link.as_path(), tgt.as_path())? {
```

Comment:
Can we add some comments here around what the casing is? ie, why check that the `__init__.py` files are the same? I'm not fully sure I follow the logic here.

Reply:
Now explained more in comments.